controllers/upgrade.go

```diff
@@ -162,10 +190,16 @@ func (r *RollingUpgradeReconciler) ReplaceNodeBatch(rollingUpgrade *v1alpha1.Rol
 			}
 			rollingUpgrade.SetLastNodeTerminationTime(metav1.Time{Time: time.Now()})
 
+			// Turns onto NodeRotationTerminate
+			rollingUpgrade.Status.NodeTurnsOntoStep(rollingUpgrade.Spec.AsgName, nodeName, v1alpha1.NodeRotationPostTerminate)
+
 			// Post Terminate Script
 			if err := r.ScriptRunner.PostTerminate(scriptTarget); err != nil {
 				return false, err
 			}
+
+			// Turns onto NodeRotationCompleted
+			rollingUpgrade.Status.NodeTurnsOntoStep(rollingUpgrade.Spec.AsgName, nodeName, v1alpha1.NodeRotationCompleted)
```

Comment:
Several of these will change once I merge my PR to parallelize the execution. We can merge this PR for now and I shall move around these calls accordingly.